libpod/container.go

```diff
@@ -688,6 +688,14 @@ func (c *Container) Terminal() bool {
 	return false
 }
 
+// LinuxResources return the contaienrs Linux Resources (if any)
+func (c *Container) LinuxResources() *spec.LinuxResources {
```

Comment:
Does this change really have positive performance impacts? I'd expect the compiler to be smart enough to allocate a variable but did not decompile.

What callers now have is one additional nil check (i.e., `if resources != nil`), so my gut feeling is performance suffered.

Reply:
Most of the callers were indeed not affected, and some may have actually been made a few cycles slower. However, the call in moveConmonToCgroupAndSignal() now avoids a call to container.Spec() which saves around 2 milliseconds (bazillions of cycles), so it does affect performance.

Now, one could add a hack just for that particular caller, but imho most of the callers that were changed to use the helper are shorter, easier to read, and generally more maintainable, while suffering negligible performance decrease.

Reply:
(8 million cycles on my machine)

Reply:
Thanks for elaborating on that.  The argument seems convincing to me. @mheon WDYT?